classes/core/AltapayOrderStatus.php

```diff
@@ -52,80 +45,77 @@ public function orderStatusChanged( $orderID, $currentStatus, $nextStatus, $orde
 		$refunded = 0;
 		$status   = '';
 
-		$api = new AltapayMerchantAPI(
-			esc_attr( get_option( 'altapay_gateway_url' ) ),
-			esc_attr( get_option( 'altapay_username' ) ),
-			esc_attr( get_option( 'altapay_password' ) ),
-			null
-		);
-		try {
-			$api->login();
-		} catch ( Exception $e ) {
-			$_SESSION['altapay_login_error'] = $e->getMessage();
+		$settings = new Core\AltapaySettings();
+		$api      = $settings->altapayApiLogin();
 
-			return new WP_Error( 'error', 'Could not login to the Merchant API: ' . $e->getMessage() );
+		if ( ! $api || is_wp_error( $api ) ) {
+			echo '<p><b>' . __( 'Could not connect to AltaPay!', 'altapay' ) . '</b></p>';
+			return;
 		}
 
-		$payment  = $api->getPayment( $txnID );
-		$payments = $payment->getPayments();
-		foreach ( $payments as $pay ) {
-			$reserved = $pay->getReservedAmount();
-			$captured = $pay->getCapturedAmount();
-			$refunded = $pay->getRefundedAmount();
-			$status   = $pay->getCurrentStatus();
+		$auth = $settings->getAuth();
+		$api  = new Payments( $auth );
+		$api->setTransaction( $txnID );
+		$payments = $api->call();
 
+		if ( $payments ) {
+			foreach ( $payments as $pay ) {
+				$reserved += $pay->ReservedAmount;
+				$captured += $pay->CapturedAmount;
+				$refunded += $pay->RefundedAmount;
+				$status   += $pay->TransactionStatus;
+			}
 		}
 
 		if ( $currentStatus === 'cancelled' ) {
 			try {
 				if ( $status === 'released' ) {
 					return;
 				} elseif ( $captured == 0 && $refunded == 0 ) {
```

Comment:
Since we know the type to be numeric we can do strict compare
```suggestion
				} elseif ( $captured === 0 && $refunded === 0 ) {
```